rq/results.py

```diff
@@ -143,16 +143,31 @@ def fetch(cls, job: Job, serializer=None) -> Optional['Result']:
         return None
 
     @classmethod
-    def fetch_latest(cls, job: Job, serializer=None) -> Optional['Result']:
-        """Returns the latest result for given job instance or ID"""
-        # response = job.connection.zrevrangebyscore(cls.get_key(job.id), '+inf', '-inf',
-        #                                           start=0, num=1, withscores=True)
-        response = job.connection.xrevrange(cls.get_key(job.id), '+', '-', count=1)
-        if not response:
-            return None
-
-        result_id, payload = response[0]
-        return cls.restore(job.id, result_id.decode(), payload, connection=job.connection, serializer=serializer)
+    def fetch_latest(cls, job: Job, serializer=None, timeout: int = 0) -> Optional['Result']:
+        """Returns the latest result for given job instance or ID.
+
+        If a non-zero timeout is provided, block for a result until timeout is reached.
+        """
+        if timeout:
+            # Unlike blpop, xread timeout is in miliseconds. "0-0" is the special value for the
+            # first item in the stream, like '-' for xrevrange.
+            timeout_ms = timeout * 1000
+            response = job.connection.xread({cls.get_key(job.id): "0-0"}, block=timeout_ms)
+            if not response:
+                return None
+            response = response[0]  # Querying single stream only.
+            response = response[1]  # Xread also returns Result.id, which we don't need.
+            result_id, payload = response[-1]  # Take most recent result.
```

Comment:
Can we use `count` argument so that `xread` only returns 1 result? https://redis.io/commands/xread/

Reply:
Unfortunately not, as `COUNT N` returns the *first* `N` results, not the latest. 

Redis doesn't have a `XREVREAD`, nor a `XREVRANGE BLOCK`